test/integ/basic.cpp

```diff
@@ -358,6 +369,68 @@ LT_BEGIN_AUTO_TEST(basic_suite, two_endpoints)
     }
 LT_END_AUTO_TEST(two_endpoints)
 
+LT_BEGIN_AUTO_TEST(basic_suite, duplicate_endpoints)
+    ok_resource ok1, ok2;
+    LT_CHECK_EQ(true, ws->register_resource("OK", &ok1));
+
+    // All of these collide and the registration fails
+    LT_CHECK_EQ(false, ws->register_resource("OK", &ok2));
+    LT_CHECK_EQ(false, ws->register_resource("/OK", &ok2));
+    LT_CHECK_EQ(false, ws->register_resource("/OK/", &ok2));
+    LT_CHECK_EQ(false, ws->register_resource("OK/", &ok2));
+
+    // Check how family interacts.
+    LT_CHECK_EQ(false, ws->register_resource("OK", &ok2, true));
+
+    // Check that switched case does the right thing, whatever that is here.
+#ifdef CASE_INSENSITIVE
+    LT_CHECK_EQ(true, ws->register_resource("ok", &ok2));
+#else
+    LT_CHECK_EQ(false, ws->register_resource("ok", &ok2));
+#endif
```

Comment:
After a bit more digging, I think these are actually backwards. And both in fact end up returning false.

tl;dr: `http_endpoint::operator <` is unconditionally case-insensitive:
https://github.com/etr/libhttpserver/blob/6d1df063501c9a7b75d68958fa3105a154a75cf0/src/details/http_endpoint.cpp#L144-L146 

That's going to be a pain to fix as it's something users could be inadvertently depending on.